subscriptions/forms.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+
+from django import forms
+from django.conf import settings
+
+
+def describe_plan(i):
+    s = settings.PRICING[i]
+    if s['calls'] == '0':
+        desc = '£%d/mth - Unlimited calls' % s['price']
+    else:
+        desc = '£%d/mth - %s calls per month' % (s['price'], s['calls'])
+    return (s['plan'], desc)
+
+
+class SubscriptionMixin(forms.Form):
+    plan = forms.ChoiceField(choices=(
+        describe_plan(i) for i in range(3)
+    ), label='Please choose a plan', widget=forms.RadioSelect)
+    charitable_tick = forms.BooleanField(
+        label='I qualify for a charitable discounted price',
+        required=False)
+    charitable = forms.ChoiceField(required=False, choices=(
```

Comment:
There was the bug I've now fixed to do with that not clearing out that information (so keeping you on free plan), so might be something in that, but even so it shouldn't get confused between type of charitable and associated data, hmm.